### ISSUE TITLE
Unable to install on Odroid C-1 (Armv7) with apt or pip

### ISSUE THREAD

**Issue by philiprenich** [streamlink/streamlink#2487]
## Bug Report

- [x] This is a bug report and I have read the contribution guidelines.

### Description

I am running ubuntu 16.04 on Armv7 architecture on an Odroid-c1. I've previously had Streamlink working on this system, but it was installed a year or so ago I believe. As I understand, from a previous issue raised, there is not a .deb package available through `apt` for Armv7 (I tried just in case, but installation failed). Instead, I attempted to use `pip` to install the package more directly. This gets a ways through the setup process, but ends up failing as well. 

I have not yet tried a manual source installation as I think I'd still need other python things to work. See below for full logs from `pip install`

### Expected / Actual behavior

A smooth-ish install using `pip` (or `apt`).
Install fails.

### Reproduction steps / Explicit stream URLs to test

1. Armv7 architecture, ubuntu 16
2. `sudo pip install streamlink`

### Log output

https://gist.github.com/philiprenich/fb05a6f3ae4809790d3c23eb2f60b12d

### Additional comments, screenshots, etc.

I'm running pip 8.1.1, and I am given a notice about upgrading to 19.1.1, but when attempting that earlier things went sideways...

Cheers for the help!

_**edit**
Don't know how I missed this, but wow! Congratulations on v1.0.0 (and beyond)!!!_

**edit 2**
corrected a note about setuptools (sentence deleted)

**Comment by gravyboat** [comment]
Your log output says libbf is missing, try installing libffi-dev and potentially python-dev.

**Comment by philiprenich** [comment]
That got things closer. It's failing on something related to `openssl` or `arm-linux-gnueabihf-gcc` now. Both are installed.

New output log: https://gist.github.com/philiprenich/f85ad307889dccfd9b34c07dceb778b1

What looks to be the relevant bits:
```
running build_ext
    generating cffi module 'build/temp.linux-armv7l-2.7/_padding.c'
    creating build/temp.linux-armv7l-2.7
    generating cffi module 'build/temp.linux-armv7l-2.7/_constant_time.c'
    generating cffi module 'build/temp.linux-armv7l-2.7/_openssl.c'
    building '_openssl' extension
    creating build/temp.linux-armv7l-2.7/build
    creating build/temp.linux-armv7l-2.7/build/temp.linux-armv7l-2.7
    arm-linux-gnueabihf-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fno-strict-aliasing -Wdate-time -D_FORTIFY_SOURCE=2 -g -fstack-protector-strong -Wformat -Werror=format-security -fPIC -I/usr/include/python2.7 -c build/temp.linux-armv7l-2.7/_openssl.c -o build/temp.linux-armv7l-2.7/build/temp.linux-armv7l-2.7/_openssl.o -Wconversion -Wno-error=sign-conversion
    build/temp.linux-armv7l-2.7/_openssl.c:498:30: fatal error: openssl/opensslv.h: No such file or directory
    compilation terminated.
error: command 'arm-linux-gnueabihf-gcc' failed with exit status 1
```

**Comment by philiprenich** [comment]
Got it! Also needed to install `libssl-dev` Are these two libraries (including `libffi-dev` @gravyboat  mentioned) expected to be on any Linux system? Is it likely because my image of Ubuntu MATE was an official Odroid and therefore possibly stripped down a bit?

tl;dr of this issue:
Using Ubuntu MATE 16.04 (from Hardkernel's odroid-c1 image) requires `libffi-dev` and `libssl-dev` to be installed (`sudo apt install libffi-dev libssl-dev`)
